lib/src/widgets/yaru_search_field.dart

```diff
@@ -0,0 +1,267 @@
+                    icon: const ClipRRect(
+                      borderRadius: suffixRadius,
+                      child: Icon(
+                        YaruIcons.edit_clear,
+                      ),
+                    ),
+                  ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  void _clear() {
+    widget.onClear?.call();
+    _controller.clear();
+  }
+}
+
+/// Combines [YaruSearchField] and [YaruSearchButton] in a [Stack]
+class YaruSearchFieldTitle extends StatefulWidget {
+  const YaruSearchFieldTitle({
+    super.key,
+    required this.searchActive,
+    required this.title,
+    this.width = 190,
+    this.titlePadding = const EdgeInsets.only(left: 45.0),
+    this.autoFocus = true,
+    this.text,
+    this.hintText,
+    this.onSubmitted,
+    this.onClear,
+    this.onSearchActive,
+    this.onChanged,
+  });
+
+  final bool searchActive;
+  final Widget title;
+  final double width;
+  final EdgeInsets titlePadding;
+  final bool autoFocus;
+  final String? text;
+  final String? hintText;
+  final void Function(String? value)? onSubmitted;
+  final void Function(String)? onChanged;
+  final void Function()? onClear;
+  final void Function()? onSearchActive;
+
+  @override
+  State<YaruSearchFieldTitle> createState() => _YaruSearchFieldTitleState();
+}
+
+class _YaruSearchFieldTitleState extends State<YaruSearchFieldTitle> {
+  late bool _searchActive;
+
+  @override
+  void initState() {
+    super.initState();
+    _searchActive = widget.searchActive;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: widget.width,
+      child: Stack(
+        alignment: Alignment.centerLeft,
+        children: [
+          if (_searchActive)
+            Center(
+              child: SizedBox(
+                height: kYaruWindowTitleBarItemHeight,
+                child: YaruSearchField(
+                  height: widget.width,
+                  hintText: widget.hintText,
+                  onClear: widget.onClear,
+                  autoFocus: widget.autoFocus,
+                  onSubmitted: widget.onSubmitted,
+                  onChanged: widget.onChanged,
+                  contentPadding: const EdgeInsets.only(
+                    bottom: 10,
+                    top: 10,
+                    right: 15,
+                    left: 45,
+                  ),
+                ),
+              ),
+            )
+          else
+            Padding(
+              padding: widget.titlePadding,
+              child: widget.title,
+            ),
+          YaruSearchButton(
+            searchActive: _searchActive,
+            onPressed: () => setState(() => _searchActive = !_searchActive),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+/// A pre-styled [YaruIconButton], ideally used in combination with [YaruSearchField]
+class YaruSearchButton extends StatelessWidget {
+  const YaruSearchButton({
+    super.key,
+    required this.searchActive,
```

Comment:
required makes sense for the search field but what if the button is used standalone?